cloud-controller-manager/osc/sets_ippermissions.go

```diff
@@ -23,135 +23,144 @@ import (
 	"encoding/json"
 	"fmt"
 
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/outscale/osc-sdk-go/v2"
 )
 
-// IPPermissionSet maps IP strings of strings to EC2 IpPermissions
-type IPPermissionSet map[string]*ec2.IpPermission
+// IPRulesSet maps IP strings of strings to OSC IpPermissions
+type IPRulesSet map[string]osc.SecurityGroupRule
 
-// IPPermissionPredicate is an predicate to test whether IPPermission matches some condition.
-type IPPermissionPredicate interface {
-	// Test checks whether specified IPPermission matches condition.
-	Test(perm *ec2.IpPermission) bool
+// NewIPRulesSetFromAWS creates a new IPRulesSet
+func NewIPRulesSetFromAWS(items ...*ec2.IpPermission) IPRulesSet {
+	s := make(IPRulesSet)
```

Comment:
Maybye use more explicit name than s, p, c, u, l2, k?

Reply:
This function will be removed in the future